Common/src/matrix_structure.cpp

```diff
@@ -975,11 +892,13 @@ void CSysMatrix::DiagonalProduct(CSysVector & vec, unsigned long row_i) {
   
 }
 
-void CSysMatrix::SendReceive_Solution(CSysVector & x, CGeometry *geometry, CConfig *config) {
+template<class ScalarType>
+template<class OtherType>
+void CSysMatrix<ScalarType>::SendReceive_Solution(CSysVector<OtherType> & x, CGeometry *geometry, CConfig *config) {
```

Comment:
#652 replaces this communication routine as well as the transpose version. We will need to update those new MPI communication routines with the templated data types you have here..